airbyte/_executors/declarative.py

```diff
@@ -49,22 +51,36 @@ def __init__(
         - If `manifest` is a dict, it will be used as is.
         """
         _suppress_cdk_pydantic_deprecation_warnings()
+
+        self.name = name
         self._manifest_dict: dict
         if isinstance(manifest, Path):
-            self._manifest_dict = cast(dict, json.loads(manifest.read_text()))
+            self._manifest_dict = cast(dict, yaml.safe_load(manifest.read_text()))
 
         elif isinstance(manifest, dict):
             self._manifest_dict = manifest
 
-        if not isinstance(self._manifest_dict, dict):
-            raise PyAirbyteInternalError(message="Manifest must be a dict.")
-
+        self._validate_manifest(self._manifest_dict)
         self.declarative_source = ManifestDeclarativeSource(source_config=self._manifest_dict)
 
         # TODO: Consider adding version detection
         # https://github.com/airbytehq/airbyte/issues/318
         self.reported_version: str | None = None
 
+    def _validate_manifest(self, manifest_dict: dict) -> None:
+        """Validate the manifest."""
+        manifest_text = json.dumps(manifest_dict)
+        if "class_name:" in manifest_text:
```

Comment:
My understanding is that this means we don't support any connector with custom components. I think this is fair but there were some talks a while back to support custom components in the Connector Builder. I don't know where those talks are at today but it could be interesting for the marketplace team to know that such change will have an impact on PyAirbyte